genie-agent/src/main/java/com/netflix/genie/agent/execution/services/impl/JobSetupServiceImpl.java

```diff
@@ -616,6 +632,12 @@ String composeScript() {
 
             sb.append(NEWLINE);
 
+            sb
+                .append("# Setup completed successfully, delete marker file").append(NEWLINE)
+                .append("rm ${").append(SETUP_ERROR_FILE_ENV_VAR).append("}").append(NEWLINE);
```

Comment:
Ha what if this delete fails? (this is a rabbit hole not actually expecting to handle every single little thing)

Not sure if `rm` behaves nicely if a file doesn't exist but if it doesn't you might want to wrap in an `if`

Reply:
I could do `rm ... || true` to make it proceed.

As is, if removal fails, the error will be captured in the setup log, and the job fails.
This seems like the correct behavior to me.

Reply:
Not sure if you’re saying if this fails the job should fail but if so I disagree cause this is just a genie internal cleanup mechanism and if it fails that really shouldn’t affect the users job run as that run can still continue fine right?